OPHD/Things/Structures/RobotCommand.cpp

```diff
@@ -53,22 +53,15 @@ void RobotCommand::addRobot(Robot* robot)
 
 
 /**
- * Removes a robot to the management pool of the Robot Command structure.
- *
- * \param	robot	Pointer to a Robot to remove from command list.
+ * Removes a robot from the management pool of the Robot Command structure.
  */
 void RobotCommand::removeRobot(Robot* robot)
 {
-	if (mRobotList.empty())
-	{
-		throw std::runtime_error("RobotCommand::removeRobot(): Robot list empty.");
-	}
+	if (!robot) { throw std::runtime_error("RobotCommand::removeRobot() called with nullptr"); }
+	if (mRobotList.empty()) { return; }
```

Comment:
If the parameter was `Robot&` rather than `Robot*`, it would be impossible to pass `nullptr` without really sick pointer casts and use of undefined behaviour. References are non-nullable. I'd recommend changing the function signature and getting rid of the `!robot` check.

----

Most methods can be written to work correctly for empty lists without explicit checks for them. I believe that is the case here. We should strive to get rid of the `mRobotList.empty()` check.

Reply:
We could do it as that but robots are all stored as pointers. I could just dereference it and pass it along that way but if using remove_if wouldn't it work fine with a nullptr anyway?

Reply:
Yes, you could add a `*` before the parameter at the call point.

You may have a point though, that `remove_if` may work with `nullptr`, so this becomes a non-issue.